pkg/cli/mapping_server.go

```diff
@@ -117,15 +129,31 @@ func (c *MappingServerCommand) RunUnstarted(ctx context.Context, args []string)
 		return nil, nil, closer, fmt.Errorf("server.NewHandler: %w", err)
 	}
 
+	srv, err := serving.New(c.cfg.Port)
+	if err != nil {
+		return nil, nil, closer, fmt.Errorf("failed to create serving infrastructure: %w", err)
+	}
+
 	closer = func() {
-		if err := handler.Cleanup(); err != nil {
-			logger.Errorw("failed to close clean up handler", "error", err)
+		var retErr error
+		successTopic.Stop()
+		failureTopic.Stop()
+		if err := processorClient.Close(); err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("failed to stop processor: %w", err))
+		}
+		if retErr != nil {
+			logger.Errorw("failed to clean up resources", "error", err)
 		}
```

Comment:
If we return the err, it will be logged. No need to log here.

Reply:
this closer function is called by `defer closer()` in [Run()](https://github.com/abcxyz/pmap/blob/main/pkg/cli/policy_server.go#L59), and when it's executed, we would need to check the error, and log the error there or change the Run() to return the error. which I assume it something we don't want to do?

Reply:
oh I misread `retErr` is defined in the closer. Why do we need it then?


```go
if err := assetClient.Close(); err != nil {
  logger.Errorw("failed to close asset inventory client: %w", err)
}
```

Reply:
I didn't find this line of code in the current code, can you provide a link to it?